packages/rich-text/src/plugins/List/index.tsx

```diff
@@ -139,7 +150,52 @@ export const withListOptions: CustomSlatePluginOptions = {
   },
 };
 
-export const createListPlugin = () =>
-  createPlateListPlugin({
+// copy of https://github.com/udecode/plate/blob/main/packages/elements/list/src/withList.ts
+// to use our custom getListInsertFragment
+const withCustomList =
+  ({ validLiChildrenTypes }: WithListOptions = {}): WithOverride =>
+  (editor) => {
+    const { insertBreak, deleteBackward, deleteForward, deleteFragment } = editor;
+
+    editor.insertBreak = () => {
+      if (getListInsertBreak(editor)) return;
+
+      insertBreak();
+    };
+
+    editor.deleteBackward = (unit) => {
+      if (getListDeleteBackward(editor, unit)) return;
+
+      deleteBackward(unit);
+    };
+
+    editor.deleteForward = (unit) => {
+      if (getListDeleteForward(editor)) return;
+
+      deleteForward(unit);
+    };
+
+    editor.deleteFragment = () => {
+      if (getListDeleteFragment(editor)) return;
+
+      deleteFragment();
+    };
+
+    editor.insertFragment = getListInsertFragment(editor);
+
+    editor.normalizeNode = getListNormalizer(editor, { validLiChildrenTypes });
+
+    return editor;
+  };
+
+export const createListPlugin = () => {
+  const options = {
     validLiChildrenTypes: LIST_ITEM_BLOCKS,
-  });
+  };
+
+  const plugin = createPlateListPlugin(options);
+
+  plugin.withOverrides = withCustomList(options);
+
+  return plugin;
+};
```

Comment:
Couldn't we instead use what we had and make minimal adjustments by some minimal overwrites?

```ts
export const createListPlugin = () =>
  createPlateListPlugin({
    validLiChildrenTypes: LIST_ITEM_BLOCKS,
    withOverrides: (options: WithListOptions) => (editor) => {
      withList(options)(editor); // This is the default for the `withOverrides` option.

      const origInsertFragments = editor.insertFragment

      editor.insertFragments = (editor: PlateEditor) => (fragment: TDescendant[]) => {
        const newFragment = fragmentToPlaintextFragment(fragment)
        origInsertFragments(newFragment)
      }
    }
  })
});
```

This should make forking `getListInsertFragment.ts` irrelevant or am I missing anything?

Reply:
That assumes we can pass a custom `withOverrides` but it looks like the list plugin [doesn't care](https://github.com/udecode/plate/blob/1406095fd388e53fcc3d7940f4b9a70125df5e36/packages/elements/list/src/createListPlugin.ts#L16) about what we pass there. cc @djagya

Reply:
Ok so it's a little bit more complicated. Returning only fragments from `fragmentToPlaintextFragment` won't solve the issue as we still need to insert these correctly (the upstream implementation doesn't handle text correctly either).

I pushed a commit attempting to reduce the amount of copied code for `withCustomList` but I don't think we can't do much about the `getListInsertFragment` for now :/

Reply:
Alright, let's just get it out then :shrug: 
I was also thinking, in the worst case let's just reduce code the way you already did.